aesara/link/jax/jax_linker.py

```diff
@@ -82,6 +94,19 @@ def thunk(
                     for jax_impl_jit in jax_impl_jits
                 ]
 
+                for n in self.fgraph.inputs:
+                    sinput = storage_map[n]
+                    if isinstance(sinput[0], RandomState) and isinstance(
```

Comment:
There should be a better way way to do this rather than checking from the `fgraph.inputs`. But I cannot make a connection with the above jax typification of `RandomVariables` (#L64-L74) of input nodes with these output nodes directly. Besides the output is basically an array which makes it even harder to detect.

Reply:
I think this is a viable approach, but we might be able to package it somewhat differently.  Since the whole thing is basically a de-JAXification (i.e. it takes JAX output and puts it into a form that the Aesara graph expects), we could establish a JAX-ify/de-JAX-ify framework for `Type`s&mdash;one that would go alongside the `jax_typify` that we already established here (e.g. we could use classes that dispatch on both "typify" and "detypify" methods).

There are two notable downsides to this approach, though: this reverse transform logic is only required by one `Type` (for now), and the logic introduces some performance overhead (i.e. we have to create `RandomState` objects just to appease Aesara's graph evaluation framework).

The only other approach that comes to mind will alleviate the latter problem, but it also introduces new logic that is comparable to this reverse transform approach.  

That approach involves changing the objects and/or logic surrounding `aesara.compile.function.types.Function` and/or `aesara.link.basic.Container` so that we can store the RNG array returned by JAX in the location that Aesara sets aside for its `RandomState`-based RNG objects.  [This](https://github.com/pymc-devs/aesara/blob/84936418af87e84322bd668c7e05e62766699906/aesara/compile/function/types.py#L1031) is where/when it tries to do that and fails.  It fails because `storage` is a `aesara.link.basic.Container` with `Container.type` equal to `RandomStateType`, so, when `storage.data = x` is attempted, it will fail when `x` doesn't have the same `Type`.

We might be able to create a subclass of `RandomStateType` that accepts `DeviceArray`s just for this purpose.  For that matter, it could even be fully functional within Aesara by performing the same conversion that you've implemented below&mdash;just within its `Type` methods.

If that's possible, we would only need to replace the `RandomStateType`s within the compiled `FunctionGraph` with this new subclass.

Reply:
Establishing a whole new de-JAX-ify framework for a single `Type` seems overkill to me now. IMO It would be better to keep this idea for when we encounter more of such classes that require de-JAX-ification.

I'll see if I can come up with something using the other approach that you've mentioned.

Reply:
Or rather what if we just make `RandomStateType` capable of accepting `Device Array` values since it seems internally it's basically just comparing types. What is the downside of doing so? (If any)